api/src/workers/base-event-handler.ts

```diff
@@ -16,101 +21,80 @@ abstract class BaseEventHandler {
-    }
-    cursors.forEach(async ({ cursor, eventName }) => {
-      let blockCursor = await cursor;
-
-      const poll = async () => {
-        let fromBlock, toBlock;
-
-        try {
-          // Calculate block range to search
-          ({ fromBlock, toBlock } = await this.getBlockRange(
-            blockCursor,
-            startingBlockHeight
-          ));
-        } catch (e) {
-          console.warn("Error retrieving block range:", e);
-        }
- 
-        if (fromBlock <= toBlock) {
+    console.log("start event polling")
+
+    const poll = async () => {
+
+      let lastestBlockHeight = await this.flowService.getLatestBlockHeight();
+      // if blockCursor does not exist in DB (1st instantiation of application), create one with lastest block height
+      let blockCursor = await this.blockCursorService.findOrCreateLatestBlockCursor(lastestBlockHeight);
+      
+      if (!blockCursor || !blockCursor.id) {
+        throw new Error("Could not get block cursor due to database error.");
+      }
+
+      // currentBlockHeight in the DB will equal the toBlock of the previous blockrange fcl query.
+      // increment fromBlock by 1 for next window.
+      let fromBlock = blockCursor.currentBlockHeight + 1;
+      let toBlock = await this.flowService.getLatestBlockHeight();
+
+      // on 1st iteration, fromBlock will be greater than toBlock due to newly inserted row.
+      if (fromBlock > toBlock) {
+        fromBlock = toBlock;
+      }
+      console.log(`Checking Block range fromBlock=${fromBlock} toBlock=${toBlock}`);
+
+      // iterate over eventNames and fetch events for block range
+      let events: any[] = [];
+      for (const eventName of this.eventNames) {
           try {
             const result = await fcl.send([
               fcl.getEventsAtBlockHeightRange(eventName, fromBlock, toBlock)
             ]);
-            const decoded = await fcl.decode(result);
-
-            if (decoded.length) {
-              decoded.forEach(async (event) => await this.onEvent(event));
-              // Record the last block that we saw an event for
-              blockCursor = await this.blockCursorService.updateBlockCursorById(
-                blockCursor.id,
-                toBlock
-              );
-            }
+            const decoded: [] = await fcl.decode(result);
+            if (decoded.length > 0) {
+              events.push(...decoded);
+            } 
           } catch (e) {
             console.error(
               `Error retrieving events for block range fromBlock=${fromBlock} toBlock=${toBlock}`,
               e
             );
           }
+      }
+
+      if (events.length > 0) {
+        // make sure events are sorted by blockHeight ascending
+        events.sort((event1, event2) => {
+          if (event1.blockHeight > event2.blockHeight) {
+            return 1;
+          } else if (event1.blockHeight < event2.blockHeight) {
+            return  -1;
+          } 
+          return 0;
+        });
```

Comment:
```suggestion
        events.sort((event1, event2) => event1.blockHeight - event2.blockHeight);
```

I think this should be equivalent

Reply:
Does this sort function keep the events sorted by name as well?

I'm worried about a case where this sort function may shuffle events within the same block. 

For example:

```js
const events = [
  { blockHeight: 6, name: "ListingAvailable },
  { blockHeight: 10, name: "ListingAvailable" },
  { blockHeight: 8, name: "ListingCompleted" },
  { blockHeight: 10, name: "ListingCompleted" }
]

console.log(sortByBlockHeight(events))

//  [
//    { blockHeight: 6, name: "ListingAvailable },
//    { blockHeight: 8, name: "ListingCompleted" },
//    { blockHeight: 10, name: "ListingCompleted" }, <--- OUT OF ORDER, available must come before complete
//    { blockHeight: 10, name: "ListingAvailable" },
//  ]
```